lang/racket/Portfile

```diff
@@ -67,6 +109,161 @@ if {${configure.build_arch} in [list ppc ppc64]} {
+                    port:libpixman \
+                    port:libpng \
+                    port:MMTabBarView \
+                    port:mpfr \
+                    port:ossp-uuid \
+                    port:pango \
+                    port:poppler
+
+    configure.args-append \
+                    --disable-origtree \
+                    --enable-macprefix \
+                    "--guibindir='${racket_apps_dir}'"
+
+    # please add at next upgrade
+    #                 --enable-nonatipkg
+
+    # please remove at next upgrade
+    # https://github.com/racket/racket/issues/5292#issuecomment-3049547317
+    patchfiles-append \
+                    patch-objc_rkt.diff \
+                    patch-guibindir.diff
+    
+    # please remove at next upgrade
+    post-configure {
+        reinplace -E \
+            "s|^\[\[:space:]]*CONFIGURE_RACKET_SO_COMPILE\[\[:space:]]*=\[\[:space:]]*$|CONFIGURE_RACKET_SO_COMPILE = PLT_CS_SLSP_SUFFIX=-nonatipkg|" \
+            ${worksrcpath}/cs/c/Makefile
+        reinplace -E \
+            "s|^\[\[:space:]]*PLT_CS_SLSP_SUFFIX\[\[:space:]]*=\[\[:space:]]*$|PLT_CS_SLSP_SUFFIX = -nonatipkg|" \
+            ${worksrcpath}/cs/c/Makefile
+    }
+    
+    pre-destroot {
+        xinstall -d ${racket_apps_dir}
+    }
+
+    post-destroot {
+        # framework
+        xinstall -d ${destroot}${frameworks_dir}
+        move        ${destroot}${prefix}/lib/${name}/${Name}.framework \
+                    ${destroot}${frameworks_dir}
+        system -W ${destroot}${frameworks_dir} \
+                    "install_name_tool -change '${Name_fw}' '@rpath/${Name_fw}' '${Name_fw}'"
+
+        # GRacket.app
+        system -W   ${destroot}${prefix}/lib/${name} \
+                    "install_name_tool -change '${prefix}/lib/${name}/${Name_fw}' '${frameworks_dir}/${Name_fw}' GRacket.app/Contents/MacOS/GRacket"
+
+        # binaries
+        foreach binfile {mzscheme racket} {
+            system -W ${destroot}${prefix}/bin \
+                    "install_name_tool -change '${prefix}/lib/${name}/${Name_fw}' '${frameworks_dir}/${Name_fw}' '${binfile}'"
+            if {${configure.build_arch} eq {arm64}} {
+                system -W ${destroot}${prefix}/bin \
+                    "codesign -f -s - '${binfile}'"
+            }
+        }
+
+        # apps
+        foreach app_name [list \
+                    DrRacket \
+                    "PLT Games" \
+                    "Racket Documentation" \
+                    Slideshow \
+            ] {
+            system -W ${destroot}${racket_apps_dir} \
+                "install_name_tool -change '${prefix}/lib/${name}/${Name_fw}' '${frameworks_dir}/${Name_fw}' '${app_name}.app/Contents/MacOS/${app_name}'"
+            if {${configure.build_arch} eq {arm64}} {
+                system "codesign -f -s - '${destroot}${racket_apps_dir}/${app_name}.app/Contents/MacOS/${app_name}'"
+            }
+        }
+    }
+}
+
+proc slash_hack {binfile build_idir replacedir} {
```

Comment:
nobody (or very few) besides you will understand what this proc is accomplishing and/or where it is used. Please explain and motivate why this is the easiest/most maintainable solution.